src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -67,7 +68,7 @@ QgsPointCloudLayer::QgsPointCloudLayer( const QString &uri,
 
 QgsPointCloudLayer::~QgsPointCloudLayer()
 {
-  // TODO: switch to using private resources for the inner workings of the stats calculation task
+  // If the task is still running we need to cancel it
```

Comment:
Right, we should cancel the task if it is running, but I guess we don't need to wait until it finished, right? (but that's more related to the other PR #48769 :-)

Reply:
We don't actually need to wait since the index is copied in the task's inner details and holds no shared data with the layer.